ludwig/modules/loss_modules.py

```diff
@@ -447,43 +462,63 @@ def weighted_sigmoid_cross_entropy(
 
 
 # used for categorical and sequence features
-def sample_values_from_classes(labels, sampler, num_classes, negative_samples,
-                               unique, class_counts, distortion):
+def sample_values_from_classes(
+    labels,
+    sampler,
+    num_classes,
+    negative_samples,
+    unique,
+    class_counts,
+    distortion,
+):
     """returns sampled_values using the chosen sampler"""
-    if sampler == 'fixed_unigram':
+    if sampler == "fixed_unigram":
         sampled_values = tf.random.fixed_unigram_candidate_sampler(
             true_classes=labels,
             num_true=1,
             num_sampled=negative_samples,
             unique=unique,
             range_max=num_classes,
             unigrams=class_counts,
-            distortion=distortion
+            distortion=distortion,
         )
-    elif sampler == 'uniform':
+    elif sampler == "uniform":
         sampled_values = tf.random.uniform_candidate_sampler(
             true_classes=labels,
             num_true=1,
             num_sampled=negative_samples,
             unique=unique,
-            range_max=num_classes
+            range_max=num_classes,
         )
-    elif sampler == 'log_uniform':
+    elif sampler == "log_uniform":
         sampled_values = tf.random.log_uniform_candidate_sampler(
             true_classes=labels,
             num_true=1,
             num_sampled=negative_samples,
             unique=unique,
-            range_max=num_classes
+            range_max=num_classes,
         )
-    elif sampler == 'learned_unigram':
+    elif sampler == "learned_unigram":
         sampled_values = tf.random.learned_unigram_candidate_sampler(
             true_classes=labels,
             num_true=1,
             num_sampled=negative_samples,
             unique=unique,
-            range_max=num_classes
+            range_max=num_classes,
         )
     else:
-        raise ValueError('Unsupported sampler {}'.format(sampler))
+        raise ValueError("Unsupported sampler {}".format(sampler))
     return sampled_values
+
+
+def rmspe_loss(targets, predictions):
+    if type(predictions) == dict and PREDICTIONS in predictions.keys():
```

Comment:
I would remove this, and just have the loss passing the right tensors and the metric passing the tensor associated with predictions key. This way this function is more atomic and reusable.

Reply:
@w4nderlust  see [commit](https://github.com/ludwig-ai/ludwig/pull/1214/commits/d93696b53d409bd297c7ae67c6cc94bccadec41b)